src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala

```diff
@@ -0,0 +1,153 @@
+  val httpMethods = List(HttpMethods.GET, HttpMethods.POST, HttpMethods.PUT,
+    HttpMethods.DELETE, HttpMethods.PATCH, HttpMethods.OPTIONS, HttpMethods.HEAD)
+
+  val uniqueId1 = "1234"
+  val uniqueId2 = "12345"
+  val uniqueId3 = "123456"
+  val exampleKey = "favoriteColor"
+  val exampleVal = "green"
+  val fullProfile = Profile(
+    fullName = Some(MockUtils.randomAlpha()),
+    email = Some(MockUtils.randomAlpha()),
+    institution = Some(MockUtils.randomAlpha()),
+    pi = Some(MockUtils.randomAlpha())
+  )
+  val incompleteProfile = Profile(fullName = Some(MockUtils.randomAlpha()))
+
+  override def beforeAll(): Unit = {
+    profileServer = startClientAndServer(MockUtils.thurloeServerPort)
+    // Generate a mock response for all combinations of profile properties and user ids
+    // to ensure that all posts to any combination will yield a successful response.
+    val properties = ProfileValidator.propertyValueMap(fullProfile) ++
+      ProfileValidator.propertyValueMap(incompleteProfile)
+    properties.keys map {
+      key =>
+        List(uniqueId1, uniqueId2, uniqueId3) map {
+          userId =>
+            profileServer
+              .when(request().withMethod("POST").withPath(
+                UserService.remoteGetKeyPath.format(userId, key)))
+              .respond(
+                org.mockserver.model.HttpResponse.response()
+                  .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
+              )
+        }
+    }
+
+    List(HttpMethods.GET, HttpMethods.POST, HttpMethods.DELETE) map {
+      method =>
+        profileServer
+          .when(request().withMethod(method.name).withPath(
+            UserService.remoteGetKeyPath.format(uniqueId1, exampleKey)))
+          .respond(
+            org.mockserver.model.HttpResponse.response()
+              .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
+          )
+    }
+    profileServer
+      .when(request().withMethod("GET").withPath(UserService.remoteGetAllPath.format(uniqueId1)))
+      .respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
+      )
+    profileServer
+      .when(request().withMethod("POST").withPath(UserService.remoteSetKeyPath))
+      .respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(MockUtils.header).withStatusCode(OK.intValue)
+      )
+  }
+
+  override def afterAll(): Unit = {
+    profileServer.stop()
+  }
+
+  "UserService" - {
+
+    "when GET-ting or DELETE-ing a single profile key" - {
+      "MethodNotAllowed response is not returned" in {
+        List(HttpMethods.GET, HttpMethods.DELETE) map {
+          method =>
+            new RequestBuilder(method)(s"/profile/$exampleKey") ~>
+              dummyUserIdHeaders(uniqueId1) ~> sealRoute(routes) ~> check {
+              log.debug(s"$method /profile/$exampleKey: " + status)
+              status shouldNot equal(MethodNotAllowed)
```

Comment:
here (and elsewhere), I'd rather see:
```
status should equal(OK)
```
... in other words, test for the specific positive case, not a single negative case.